src/processes.c

```diff
@@ -1529,8 +1529,14 @@ static char *ps_get_owner(pid_t pid)
         uid = atoi (line + 5);
         getpwuid_r (uid, &passwd, passwd_buffer, sizeof(passwd_buffer),
                 &passwd_result);
-        if (passwd_result)
+        if (passwd_result) {
             result = sstrdup (passwd_result->pw_name);
+        } else {
+	    // Send the numeric uid if name is not available.
+            char uid_str[6];  // 5 digits plus string terminator.
+            snprintf(uid_str, sizeof(uid_str), "%u", uid);
```

Comment:
As a technical matter, ```%u``` is not the right format specifier for a variable of type ```int```.

I think, due to the issue I raised above (the code not knowing the size of user ids), I think I'd recommend using a bigger type (namely ```unsigned long```) in line 1520, then using ```strtoul``` rather than atoi in line 1529, and then using ```%ul``` here.

I will also note how funny it looks to take a string, parse it into an integer, and then convert that integer back to a string. But that's probably tolerable because it's the shortest version of this code.